File: group5.ts

abstract class InputDevice {
    abstract input(): void;
  }
  
  class Keyboard extends InputDevice {
    input(): void {
      console.log("Inputting data from the keyboard...");
    }
  }
  
  class Mouse extends InputDevice {
    input(): void {
      console.log("Inputting data from the mouse...");
    }
  }
  
  class Computer {
    private inputDevice: InputDevice;
  
    constructor(inputDevice: InputDevice) {
      this.inputDevice = inputDevice;
    }
  
    input(): void {
      this.inputDevice.input();
    }
  
    wifiConnection(): any {
      console.log('Discoverable... Connect');
      return {};
    }
  
    setInputDevice(inputDevice: InputDevice): void {
      this.inputDevice = inputDevice;
    }
  
    getInputDevice(): InputDevice {
      return this.inputDevice;
    }
  }
  
  const computer1 = new Computer(new Keyboard());
  
  function testComputer(computer: Computer): void {
    computer.input();
    computer.wifiConnection();
  }
  
  testComputer(computer1);
  
  interface WifiConnect {
    wifiConnection(): string;
  }
  
  interface Bluetooth {
    blueTooth(): void;
  }
  
  class Smartphone implements WifiConnect, Bluetooth {
    wifiConnection(): string {
      return "Connecting";
    }
  
    blueTooth(): void {
      console.log("Bluetooth is connected.");
      console.log("Sending a file via Bluetooth...");
    }
  }
  
  class Laptop implements WifiConnect, Bluetooth {
    wifiConnection(): string {
      console.log("Enter password");
      return "Connecting";
    }
  
    blueTooth(): void {
      console.log("Bluetooth is connected.");
      console.log("Receiving a file via Bluetooth...");
    }
  }
  
  class walltop implements WifiConnect {
    wifiConnection(): string {
      console.log("Connecting");
      return "Connecting";
    }
  }
  
  class SmartWatch implements WifiConnect, Bluetooth {
    wifiConnection(): string {
      console.log("Connecting");
      return "Connecting";
    }
  
    blueTooth(): void {
      console.log("Bluetooth is connected.");
      console.log("Monitoring heart rate via Bluetooth...");
    }
  }
  
  
  class ButtonPhone extends InputDevice {
    input(): void {
      console.log("Inputting data from Button Phone");
    }
  }
  

  const buttonPhone = new ButtonPhone();
  computer1.setInputDevice(buttonPhone);
  testComputer(computer1);
  
  const smartphone = new Smartphone();
  smartphone.blueTooth(); 
  
  const laptop = new Laptop();
  laptop.blueTooth();  
  // "Bluetooth is connected." 
  
  const smartwatch = new SmartWatch();
  smartwatch.blueTooth(); 
  //  "Bluetooth is connected." 
  


  

File: group5.py
from abc import ABC, abstractmethod

class InputDevice(ABC):
    @abstractmethod
    def input(self):
        pass

class Keyboard(InputDevice):
    def input(self):
        print("Inputting data from the keyboard...")

class Mouse(InputDevice):
    def input(self):
        print("Inputting data from the mouse...")

class Computer:
    def __init__(self, input_device):
        self.__input_device = input_device

    def input(self):
        self.__input_device.input()

    def wifi_connection(self):
        print('Discoverable... Connect')
        return {}

    def set_input_device(self, input_device):
        self.__input_device = input_device

    def get_input_device(self):
        return self.__input_device

computer1 = Computer(Keyboard())

def test_computer(computer):
    computer.input()
    computer.wifi_connection()

test_computer(computer1)

class WifiConnect(ABC):
    @abstractmethod
    def wifi_connection(self):
        pass


class Bluetooth(ABC):
    @abstractmethod
    def blue_tooth(self):
        pass

class Smartphone(WifiConnect, Bluetooth):
    def wifi_connection(self):
        return "Connecting"

class Laptop(WifiConnect,Bluetooth):
    def wifi_connection(self):
        print("Enter password")




class RaspberryPi(WifiConnect):
    def wifi_connection(self):
        print("Connecting")

class SmartWatch(WifiConnect,Bluetooth):
    def wifi_connection(self):
        print("Connecting")

# Update ButtonPhone to match the InputDevice interface
class ButtonPhone(InputDevice):
    def input(self):
        print("Inputting data from Button Phone")

# Create a ButtonPhone instance
button_phone = ButtonPhone()
computer1.set_input_device(button_phone)
test_computer(computer1)
